openspec/changes/extract-shared-base-cache-code/specs/shared-admission-logic/spec.md

```
@@ -0,0 +1,71 @@
+- **OR** insufficient victims are available
+- **THEN** returns `AdmissionResult::Rejected`
+
+### Requirement: handle_admit SHALL update deques synchronously
+
+`handle_admit` SHALL update the deques and counters without acquiring locks:
+- Increment entry and weight counters
+- Push entry to access-order deque
+- Push entry to write-order deque (if enabled)
+- Set entry's admitted flag
+
+#### Scenario: Entry admitted to deques
```

markdownlint-cli2 (0.21.0)

[warning] Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)